stdlib/sort.jou

```diff
@@ -0,0 +1,81 @@
+# This file lets you sort an array of values.
+#
+# See doc/sorting.md for documentation.
+
+import "stdlib/str.jou"
+
+declare qsort(
+    base: void*,
+    nmemb: int64,
+    size: int64,
+    compare: funcptr(void*, void*) -> int,
+) -> None
+
+
+# The "advanced" way to sort elements. You provide the data to sort and a
+# comparator function.
+#
+# The comparator takes two pointers to values in the array and returns an
+# integer. The return value should be:
+#
+#   negative    if first < second
+#   0           if first == second
+#   positive    if first > second
+@public
+class Sorter[T]:
+    @inline
+    def sort(self: Sorter[T], ptr: T*, len: int64, compare: funcptr(T*, T*) -> int) -> None:
+        qsort(ptr, len, sizeof(ptr[0]), compare as funcptr(void*, void*) -> int)
```

Comment:
Should probably rename `compare` to `comparator`, because that's what docs call it.